markdown/1-Introduction/exercises/ex_1/index.md

```diff
@@ -4,68 +4,6 @@ title: Exercise 1.1
 permalink: /intro-exercises/ex_1/
 breadcrumb: 1-Introduction
 breadcrumb2: ex_1
```

Comment:
@sachin10101998 Breadcrumbs represents a hierarchical level of hyperlinks that are displayed on the top of a web page. They facilitate navigation. The word 'hierarchy' here refers to the level that a certain exercise belongs to (excluding its current level). Thus, the only thing that'll go into an exercise's breadcrumb is the 'Chapter' that contains the exercise. 
Using the keys breadcrumb2, breadcrumb3, breadcrumb4, and breadcrumb5 implies that the hierarchy keeps on increasing (which obviously is not the case). Bottom line is, you **cannot** use the word "breadcrumb#" to represent a concept that is facilitating some other task (like linking a configuration or ). 

Now coming to the purpose of these 'keys' in front-matter (that you decided to name breadcrumb#). You seem to have used values from these 'keys' as follows:

**breadcrumb3 and breadcrumb2 are being used in the layout file of exercises for fetching Github API links.**
```
var chapterName = String('{{ page.breadcrumb }}');
var exerciseName = String('{{ page.breadcrumb2 }}');
var chapterNamestripped = String('{{ page.breadcrumb3 }}');
```
exerciseName is not a 'breadcrumb'. If you need something to uniquely identify an exercise, then use something like 'canonical_name' instead of 'breadcrumb2'. 
For breadcrumb3, did you literally create a separate metadata field to get 'stripped' names? You are using these values in Javascript   and a single line could have 'stripped' hyphens from a string. WHY?

**breadcrumb4 and breadcrumb5 are similar and seem to be used by staticman.**
If they're similar, you don't need them both. Also, have you considered clubbing this with any other identifier in front-matter (maybe 'canonical_name' that I mentioned above)?

These redundant breadcrumbs are "inside the front-matter" that represents the meta-data of an exercise. You have to name them properly and avoid duplications.

Reply:
Breadcrumb3 aka chapterNamestripped and breadcrumb4 (used in staticman) have been removed totally. Will send another commit by renaming the `Breadcrumb2` with `exerciseNumber` and beadcrumb5 by `bookmarkID`.

Reply:
@Nalinc done.